src/lib/trace-ir/field-class.c

```diff
@@ -669,6 +671,23 @@ add_mapping_to_enumeration_field_class(struct bt_field_class *fc,
 	BT_LIB_LOGD("Added mapping to enumeration field class: "
 		"%![fc-]+F, label=\"%s\"", fc, label);
 
+	// See if this remains a bit field enum field
+	if (enum_fc->is_bit_field_enum) {
+		if (range_set->ranges->len != 1) {
+			enum_fc->is_bit_field_enum = false;
+		} else {
+			const struct bt_integer_range *range = (const void *)
+				BT_INTEGER_RANGE_SET_RANGE_AT_INDEX(
+					range_set, 0);
+			if ((range->lower.i != range->upper.i) ||
+				!(range->lower.i > 0 && (range->lower.i & (range->lower.i - 1)) == 0)) {
```

Comment:
Do we really want to use lower.i here ? And why exclude values with the 64-th bit set (which appear negative) ? I suspect we want to deal with lower.u and upper.u instead, and don't care about sign.